File: ejemplo/src/app/model/pedido-request.ts
import { Client } from './client.model';
import { Inventory } from './inventory.model';
import { Pedido } from './pedido.model';
import { PedidoDetalle } from './pedidoDetalle.model';

export interface PedidoRequest {
  pedido: Pedido;
  productos: Inventory[];
  cantidades: number[];
}

File: Inventory/src/main/java/com/inventory/manager/model/modelRequest/PedidoRequest.java
package com.inventory.manager.model.modelRequest;

import com.inventory.manager.model.Pedido;
import com.inventory.manager.model.PedidoDetalle;
import lombok.Data;

@Data
public class PedidoRequest {
    private Pedido pedido;
    private PedidoDetalleRequest pedidoDetalleRequest;
}
